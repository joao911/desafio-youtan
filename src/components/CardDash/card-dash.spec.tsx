import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { CardDash } from "./index";

describe("CardDash Component", () => {
  it("renders correctly", () => {
    const { getByText } = render(
      <CardDash icon={<div>Icon</div>} title={100} subtitle="Subtitle" />
    );
    expect(getByText("Subtitle")).toBeDefined();
  });
  it("renders the icon correctly", () => {
    const { container } = render(
      <CardDash
        icon={<div data-testid="icon">Icon</div>}
        title={100}
        subtitle="Test Subtitle"
      />
    );
    expect(container.querySelector('[data-testid="icon"]')).toBeInTheDocument();
  });

  it("renders the subtitle correctly", () => {
    const { getByText } = render(
      <CardDash icon={<div>Icon</div>} title={100} subtitle="My Subtitle" />
    );
    expect(getByText("My Subtitle")).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <CardDash icon={<div>Icon</div>} title={100} subtitle="Snapshot Test" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with title as zero", () => {
    const { getByText } = render(
      <CardDash icon={<div>Icon</div>} title={0} subtitle="Test Subtitle" />
    );
    expect(getByText("0")).toBeInTheDocument();
  });
});
