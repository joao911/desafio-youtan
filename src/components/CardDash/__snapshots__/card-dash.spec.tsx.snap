// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardDash Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root w-full p-4 css-1j74uew-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="flex gap-4"
    >
      <div>
        Icon
      </div>
      <div>
        <p
          class="text-xl font-bold text-text-secondary"
        >
          <span>
            0
          </span>
        </p>
        <p
          class="text-text-subtitle"
        >
          Snapshot Test
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
